File: src/main/java/com/wellbeignatwork/backend/entity/Collaboration/Reservation.java
package com.wellbeignatwork.backend.entity.Collaboration;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.wellbeignatwork.backend.entity.User.User;
import lombok.*;
import lombok.experimental.FieldDefaults;
import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@ToString
@Entity
public class Reservation implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long idReservation;
    LocalDateTime startDateRes;
    LocalDateTime endDateRes;
    int nmPalce;
    int priceTotal;


    @JsonIgnore
    @ManyToOne
    Offer offersRes;
    @JsonIgnore
    @ManyToOne
    User userRes;

}

File: src/app/Models/Collaboration/Reservation.ts
import { User } from './user';
import { Offer } from './offer';
import { Image } from './Image';
export class Reservation {
    idReservation: Number;  
    startDateRes:Date; 
    endDateRes:Date;
    nmPalce:Number;
    priceTotal:Number;
    userRes :User;
    offersRes? : Offer;
}